src/custom/hooks/usePriceImpact/index.ts

```diff
@@ -123,24 +132,41 @@ interface AbaPriceImpactParams {
   fiatPriceImpact?: Percent
 }
 
-function _calculateAbaPriceImpact(abTrade: TradeGp, baTrade: TradeGp, isExactIn: boolean) {
-  console.debug('abTrade', abTrade, 'baTrade', baTrade, isExactIn)
+function _calculateAbaPriceImpact(initialValue: string, finalValue: string) {
+  const initialValueBn = new BigNumber(initialValue)
+  const finalValueBn = new BigNumber(finalValue)
+  // TODO: use correct formula
+  // ((IV - FV) / IV / 2) * 100
+  const [numerator, denominator] = initialValueBn.minus(finalValueBn).div(initialValueBn).div('2').toFraction()
+
+  // we don't want zeroes
+  if (numerator.isZero() || denominator.isZero()) return undefined
+
+  const priceImpactPercentage =
+    // Uni sdk hates negative numbers so we need to do this
+    numerator.isNegative() || denominator.isNegative()
+      ? new Percent(numerator.absoluteValue().toString(10), denominator.absoluteValue().toString(10)).multiply('-1')
+      : new Percent(numerator.toString(10), denominator.toString(10))
 
-  return HARD_CODED_PRICE_IMPACT
+  console.debug(`[useAbaPriceImpact]::${priceImpactPercentage.toSignificant(2)}%`)
+
+  return priceImpactPercentage
 }
 
 export function useAbaPriceImpact({ abTrade, fiatPriceImpact }: AbaPriceImpactParams) {
   const {
     INPUT: { currencyId: sellToken },
     OUTPUT: { currencyId: buyToken },
+    independentField,
   } = useSwapState()
-  const isExactIn = abTrade?.tradeType === TradeType.EXACT_INPUT
+  const isExactIn = independentField === Field.INPUT
   // we calculate the trade going B > A
   // using the output values from the original A > B trade
   const baTrade = useQuoteAndSwap({
+    // if priceImpact, give undefined and dont compute swap
     // the amount traded now is the A > B output amount without fees
     // TODO: is this the amount with or without fees?
-    parsedAmount: isExactIn ? abTrade?.outputAmount : abTrade?.inputAmount,
+    parsedAmount: !fiatPriceImpact ? (isExactIn ? abTrade?.outputAmount : abTrade?.inputAmount) : undefined,
```

Comment:
nit1: declare const for the arguments of `useQuoteAndSwap` to make the call simpler?

nit2: Why not `fiatPriceImpact ? undefined : formula`. Also, not super fan of chaining ternary operators